reporters/index.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+var getReporter = require('jscs/lib/cli-config').getReporter;
+var PluginError = require('gulp-util').PluginError;
+var through = require('through2');
+
+exports.failReporter = require('./fail');
+
+exports.loadReporter = function (reporter) {
```

Comment:
JSCS already exposes `getReporter()`.
This `loadReporter` function simply wraps JSCS's `getReporter()` to add support for loading non-built-in reporters (e.g. passing a function or a reporter package name installed from npm).

Taking a second look at this, I've removed the "object reporters" -- `{ reporter: function(){} }` -- that was plainly redundant as you can just pass a function directly.

Also, it looks like the logic for loading reporters from npm packages is not correct at all, I'll add some unit tests regarding reporters.